src/test/scala/com/cognite/sdk/scala/common/StringDataPointsResourceBehaviors.scala

```diff
@@ -59,9 +69,26 @@ trait StringDataPointsResourceBehaviors extends Matchers with RetryWhile { this:
         )
 
         dataPoints.deleteRangeById(stringTimeSeriesId, start, end.plusMillis(1))
-        retryWithExpectedResult[Seq[StringDataPointsByIdResponse]](
+        retryWithExpectedResult[StringDataPointsByExternalIdResponse](
           dataPoints.queryStringsByExternalId(stringTimeSeriesExternalId, start, end.plusMillis(1)),
-          pad => pad.head.datapoints should have size 0
+          pad => pad.datapoints should have size 0
         )
     }
+
+    it should "support support query by externalId when ignoreUnknownIds=true" in {
+      val doesNotExist = "does-not-exist-" + UUID.randomUUID
+      dataPoints.getLatestStringDataPointByExternalIds(Seq(doesNotExist), ignoreUnknownIds = true) shouldBe empty
+      dataPoints.getLatestDataPointsByExternalIds(Seq(doesNotExist), ignoreUnknownIds = true) shouldBe empty
+      dataPoints.queryByExternalIds(Seq(doesNotExist), Instant.EPOCH, Instant.now, ignoreUnknownIds = true) shouldBe empty
+      dataPoints.queryStringsByExternalIds(Seq(doesNotExist), Instant.EPOCH, Instant.now, ignoreUnknownIds = true) shouldBe empty
+    }
+
+    it should "support support query by internal id when ignoreUnknownIds=true" in {
+      val doesNotExist = ThreadLocalRandom.current().nextLong(1, 9007199254740991L)
```

Comment:
Easier to just say `val doesNotExist = 123L`, or `9007199254740991L`, because it won't. ;)